src/AspNetCore/Src/applicationinfo.cpp

```diff
@@ -146,12 +155,22 @@ APPLICATION_INFO::UpdateAppOfflineFileHandle()
                     strEventMsg.QueryStr());
             }
 
-            m_pApplication->ShutDown();
-            m_pApplication->DereferenceApplication();
-            m_pApplication = NULL;
+            APPLICATION_MANAGER::GetInstance()->RecycleApplicationFromManager(
+                m_pApplication->QueryConfig()->QueryConfigPath()->QueryStr()
+            );
 
+            if (m_pConfiguration->QueryHostingModel() == APP_HOSTING_MODEL::HOSTING_OUT_PROCESS)
```

Comment:
This can be put into RecycleApplication if you like

Reply:
Not moving this for now, but I think there a few potential issues with app_offline that I would like to address first.